spidermon/contrib/stats/statscollectors.py

```diff
@@ -0,0 +1,35 @@
+import os
+import pickle
+from collections import deque
+
+from scrapy.statscollectors import StatsCollector
+from scrapy.utils.project import data_path
+
+
+class LocalStorageStatsHistoryCollector(StatsCollector):
+    def __init__(self, crawler):
+        super(LocalStorageStatsHistoryCollector, self).__init__(crawler)
+
+        self._statsdir = data_path("stats", createdir=True)
+
+    def open_spider(self, spider):
+        stats_location = os.path.join(
+            self._statsdir, "{}_stats_history".format(spider.name)
+        )
+        if os.path.isfile(stats_location):
+            with open(stats_location, "rb") as stats_file:
+                _stats_history = pickle.load(stats_file)
+        else:
+            max_stored_stats = spider.crawler.settings.getint(
+                "SPIDERMON_MAX_STORED_STATS", default=100
+            )
+            _stats_history = deque([], maxlen=max_stored_stats)
+        spider.stats_history = _stats_history
+
+    def _persist_stats(self, stats, spider):
+        spider.stats_history.appendleft(self._stats)
+        stats_location = os.path.join(
+            self._statsdir, "{}_stats_history".format(spider.name)
+        )
```

Comment:
I see some repetition here and in `open_spider` that may be worth avoiding. Maybe `stats_location` could be defined on the constructor, possibly making `self._statsdir` unnecessary? I think that making `statsdir` an optional parameter to the constructor as well should still allow for the tests to override the directory with their own temporary directory.

Reply:
I also think we can move that bit of code to a separated function like `get_stats_location`, this way is also easy to overload this in the future if needed.

Reply:
I removed the code duplication. It makes no sense to add it to the constructor as it will never be called except for the tests.